p2p_wallet/AppDelegate+Injection.swift

```diff
@@ -157,6 +254,56 @@ extension Resolver: ResolverRegistering {
             .implements(BuyProviderFactory.self)
             .scope(.application)
         
+        // MARK: - TransactionInfo
+        register { TransactionInfoViewModel() }
+            .scope(.shared)
+        
+        // MARK: - BuyRoot
+        register { BuyRoot.ViewModel() }
+            .implements(BuyViewModelType.self)
+            .scope(.shared)
+        
+        register { SolanaBuyToken.SceneModel() }
+            .implements(SolanaBuyTokenSceneModel.self)
+            .scope(.shared)
+        
+        // MARK: - Receive
+        register { ReceiveToken.SceneModel() }
+            .implements(ReceiveSceneModel.self)
+            .scope(.shared)
+        
+        // MARK: - Send
+        register { SendToken.ViewModel() }
+            .implements(SendTokenViewModelType.self)
+            .scope(.shared)
+        
+        // MARK: - OrcaSwap
+        register { OrcaSwapV2.ViewModel() }
```

Comment:
I think having a central place for provider or services injection is good, but not for view model injection. How about to inject them in scene level? We can probably use some code generators like `Sourcery` to do it.

```swift
struct SomeScene {
   enum Navigatable {...}
   static func inject() {
      register { ViewModel() }.scope(.shared) // SomeScene.ViewModel
   }
}
```

Reply:
apparently, the register function must be called in registerAllServices()

Reply:
You are right about `registerAllServices()`. I mean we can probably use `Sourcery` to generate the code, that automatically call inject function in each Scene from `registerAllServices()`.

Reply:
Hi there! Sound great! Do you have any article talking about this?

Reply:
Here is an [example](https://atomicrobot.com/blog/sourcery/) :D